frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/OpenMacheteFileAction.java

```diff
@@ -34,23 +33,25 @@ public LambdaLogger log() {
   @Override
   @UIEffect
   public void actionPerformed(AnActionEvent anActionEvent) {
-    Project project = getProject(anActionEvent);
+    val project = getProject(anActionEvent);
 
     // When selected Git repository is empty (due to e.g. unopened Git Machete tab)
     // an attempt to guess current repository based on presently opened file
-    val gitDir = getSelectedGitRepository(anActionEvent)
-        .onEmpty(() -> DvcsUtil.guessCurrentRepositoryQuick(project,
+    val selectedGitRepo = getSelectedGitRepository(anActionEvent);
+    val repo = selectedGitRepo != null
+        ? selectedGitRepo
+        : DvcsUtil.guessCurrentRepositoryQuick(project,
             GitUtil.getRepositoryManager(project),
-            GitVcsSettings.getInstance(project).getRecentRootPath()))
-        .map(GitVfsUtils::getMainGitDirectory);
+            GitVcsSettings.getInstance(project).getRecentRootPath());
+    val gitDir = repo != null ? GitVfsUtils.getMainGitDirectory(repo) : null;
```

Comment:
Nit: use `@ExtensionMethod` and call as
```suggestion
    val gitDir = repo != null ? repo.getMainGitDirectory() : null;
```